bot/utils/randomization.py

```diff
@@ -0,0 +1,19 @@
+import itertools
+import random
+import typing as t
+
+
+class RandomCycle:
+    """Cycling through jumping to random indexes in an iterable."""
```

Comment:
I think this could have a more descriptive docstring, such as:
```suggestion
    """
    Cycles through elements from a randomly shuffled iterable, repeating indefinitely.

    The iterable is reshuffled after each full cycle.
    """
```